include/zephyr/drivers/sensor.h

```diff
@@ -691,6 +691,89 @@ static inline int sensor_value_from_double(struct sensor_value *val, double inp)
 	return 0;
 }
 
+struct sensor_info {
+	const struct device *dev;
+	const char *vendor;
+	const char *model;
+	const char *friendly_name;
+};
+
+#ifdef CONFIG_SENSOR_INFO
+
+#define SENSOR_INFO_INITIALIZER(_dev, _vendor, _model, _friendly_name)	\
+	{								\
+		.dev = _dev,						\
+		.vendor = _vendor,					\
+		.model = _model,					\
+		.friendly_name = _friendly_name,			\
+	}
+
+#define SENSOR_INFO_DEFINE(name, ...)					\
+	static const STRUCT_SECTION_ITERABLE(sensor_info, name) =	\
+		SENSOR_INFO_INITIALIZER(__VA_ARGS__)
+
+#define SENSOR_INFO_NAME(node_id)					\
+	_CONCAT(__sensor_info, DEVICE_DT_NAME_GET(node_id))
+
+#else
+
+#define SENSOR_INFO_DEFINE(name, ...)
+#define SENSOR_INFO_NAME(node_id)
+
+#endif /* CONFIG_SENSOR_INFO */
+
+/**
+ * @brief Like DEVICE_DT_DEFINE() with sensor specifics.
+ *
+ * @details Defines a device which implements the sensor API. May define an
+ * element in an iterable section be used by the sensor subsystem to identify
+ * all sensor devices.
+ *
+ * @param node_id The devicetree node identifier.
+ *
+ * @param init_fn Name of the init function of the driver.
+ *
+ * @param pm_device PM device resources reference (NULL if device does not use
+ * PM).
+ *
+ * @param data_ptr Pointer to the device's private data.
+ *
+ * @param cfg_ptr The address to the structure containing the configuration
+ * information for this instance of the driver.
+ *
+ * @param level The initialization level. See SYS_INIT() for details.
+ *
+ * @param prio Priority within the selected initialization level. See
+ * SYS_INIT() for details.
+ *
+ * @param api_ptr Provides an initial pointer to the API function struct used
+ * by the driver. Can be NULL.
+ */
+#define SENSOR_DEVICE_DT_DEFINE(node_id, init_fn, pm_device,		\
```

Comment:
I'd like to see us go further here, see my PR for boilerplate UART_DT_INST_DEFINE

Reply:
@galak can you include the link to the PR?

Reply:
@yperess https://github.com/zephyrproject-rtos/zephyr/pull/49295